envoy/admin/v2/admin.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package envoy.admin;
+
+import "google/protobuf/any.proto";
+import "envoy/api/v2/rds.proto";
+
+import "gogoproto/gogo.proto";
+
+// [#protodoc-title: ConfigDump]
+
+// The /config_dump admin endpoint uses this wrapper message to maintain and serve arbitrary
+// configuration information from any component in Envoy.
+message ConfigDump {
+  // This map is serialized and dumped in its entirety at the /config_dump endpoint.
+  //
+  // Keys should be a short descriptor of the config object they map to. For example, envoy's HTTP
+  // routing subsystem might use "routes" as the key for its config, for which it uses the message
+  // RouteConfigDump as defined below. In the future, the key will also be used to filter the output
+  // of the /config_dump endpoint.
+  map<string, google.protobuf.Any> configs = 1 [(gogoproto.nullable) = false];
+}
+
+// Envoy's RDS implementation fills this message with all currently loaded routes, as described by
+// their RouteConnfiguration objects. Static routes configured in the bootstrap configuration are
+// separated from those configured dynamically via RDS. This message is available at the
+// /config_dump admin endpoint.
+message RouteConfigDump {
```

Comment:
This is a good point; we want to be able to dump static and dynamic information. What if we have an endpoint to dump the entire bootstrap? Would that be sufficient for covering the static requirements?

Reply:
Yes we should definitely have a bootstrap dump option. I could go either way on whether that covers though, because for example a route table can come as part of a static LDS route. I think it's probably OK for "routes" to dump both static and dynamic.

Reply:
Yes, the Bootstrap should definitely have a dump. Is that the sort of thing I should be opening issues for? I'm keeping notes locally so I don't forget.

More to the point, it seems helpful to have this distinction available without going for the bootstrap dump. IMO RouteConfiguration could use a flag indicating its static/dynamic origin but maybe that ship has sailed.